ldap/servers/slapd/back-ldbm/dbimpl.c

```diff
@@ -397,7 +397,48 @@ const char *dblayer_op2str(dbi_op_t op)
     return str[idx];
 }
 
-/* Open db env, db and db file privately */
+/* Get the li_directory directory from the database instance name -
+ * Caller should free the returned value
+ */
+static char *
+get_li_directory(const char *fname)
+{
+    /*
+     * li_directory is an existing directory.
+     * it can be fname or its parent or its greatparent
+     * in case of problem returns the provided name
+     */
+    char *lid = slapi_ch_strdup(fname);
+    struct stat sbuf = {0};
+    char *pt = NULL;
+    for (int i=0; i<3; i++) {
```

Comment:
Minor `size_t` for array indexing issue

Reply:
But in this case i is not an array index ..., 
It is a count to know how many time we try to look for parent directories 
==> will keep int but rename i as count for better clarity